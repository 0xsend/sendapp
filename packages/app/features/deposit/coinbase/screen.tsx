import { useState } from 'react'
import { OnrampFlow } from 'app/features/deposit/components/OnrampFlow'
import { useSendAccount } from 'app/utils/send-accounts'
import { useCoinbaseOnramp } from 'app/utils/useCoinbaseOnramp'
import { toNiceError } from 'app/utils/toNiceError'
import { YStack, Text, Button, Spinner } from '@my/ui'
import { CoinbaseOnrampVerifyScreen } from '../components/CoinbaseOnrampVerifyScreen'
import { useRouter } from 'solito/router'

const COINBASE_APP_ID = process.env.NEXT_PUBLIC_CDP_APP_ID ?? ''

interface DepositCoinbaseScreenProps {
  defaultPaymentMethod?: 'APPLE_PAY' | 'CARD'
}

export function DepositCoinbaseScreen({ defaultPaymentMethod }: DepositCoinbaseScreenProps) {
  const router = useRouter()
  const { data: sendAccount } = useSendAccount()
  const {
    openOnramp,
    closeOnramp,
    status: coinbaseStatus,
    error,
    isLoading,
  } = useCoinbaseOnramp({
    projectId: COINBASE_APP_ID,
    address: sendAccount?.address ?? '',
    partnerUserId: sendAccount?.user_id ?? '',
    defaultPaymentMethod,
  })

  // Track transaction status
  const [status, setStatus] = useState<'idle' | 'failure'>('idle')

  const handleConfirmTransaction = (amount: number) => {
    openOnramp(amount)
  }

  useEffect(() => {
    if (coinbaseStatus === 'success') {
      router.push('/deposit/success')
    }
  }, [coinbaseStatus, router.push])

  const renderContent = () => {
    switch (true) {
      case !!error:
        return (
          <YStack ai="center" gap="$4" py="$8">
            <Text fontSize="$6" fontWeight="500" color="$red10" ta="center">
              Onramp page closed
            </Text>
            <Text color="$gray11" ta="center">
              {toNiceError(error)}
            </Text>
            <Button
              variant="outlined"
              color="$color"
              size="$4"
              hoverStyle={{
                backgroundColor: '$color1',
                borderColor: '$color8',
              }}
              pressStyle={{
                backgroundColor: '$color2',
              }}
              onPress={closeOnramp}
            >
              Try Again
            </Button>
          </YStack>
        )

      case coinbaseStatus === 'success':
        return (
          <YStack ai="center" gap="$4" py="$8">
            <Spinner size="large" color="$primary" />
            <Text fontSize="$6" fontWeight="500" ta="center">
              Transaction Complete
            </Text>
            <Text color="$gray11" ta="center">
              Finishing up...
            </Text>
          </YStack>
        )

      case coinbaseStatus === 'payment_submitted':
        return (
          <CoinbaseOnrampVerifyScreen
            onFailure={() => setStatus('failure')}
            onSuccess={() => router.push('/deposit/success')}
          />
        )

      case coinbaseStatus === 'pending_payment':
        return (
          <YStack ai="center" gap="$4" py="$8">
            <Spinner size="large" color="$primary" />
            <Text fontSize="$6" fontWeight="500" ta="center">
              Processing Transaction
            </Text>
            <Text color="$gray11" ta="center">
              Complete in Coinbase window
            </Text>
            <Button
              variant="outlined"
              color="$color"
              size="$4"
              hoverStyle={{
                backgroundColor: '$color1',
                borderColor: '$color8',
              }}
              pressStyle={{
                backgroundColor: '$color2',
              }}
              onPress={closeOnramp}
            >
              Cancel
            </Button>
          </YStack>
        )

      case status === 'failure':
        return (
          <YStack ai="center" gap="$4" py="$8">
            <Text fontSize="$6" fontWeight="500" color="$red10" ta="center">
              Transaction Timed Out
            </Text>
            <Text color="$gray11" ta="center">
              Your payment took too long to process. Please try again.
            </Text>
            <Button variant="outlined" color="$color" size="$4" onPress={closeOnramp}>
              Try Again
            </Button>
          </YStack>
        )

      case coinbaseStatus === 'failed':
        return (
          <YStack ai="center" gap="$4" py="$8">
            <Text fontSize="$6" fontWeight="500" color="$red10" ta="center">
              Transaction Failed
            </Text>
            <Text color="$gray11" ta="center">
              Your payment could not be processed. Please try again.
            </Text>
            <Button variant="outlined" color="$color" size="$4" onPress={closeOnramp}>
              Try Again
            </Button>
          </YStack>
        )

      default:
        return <OnrampFlow onConfirmTransaction={handleConfirmTransaction} isLoading={isLoading} />
    }
  }

  return (
    <YStack mt="$4" mx="auto" width={'100%'} $sm={{ maxWidth: 600 }}>
      <YStack w={'100%'}>
        <YStack f={1} px="$4" jc="space-between" pb="$4">
          {renderContent()}
        </YStack>
      </YStack>
    </YStack>
  )
}
