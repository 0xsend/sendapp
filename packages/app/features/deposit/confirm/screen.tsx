import { useSendAccount } from 'app/utils/send-accounts'
import { useCoinbaseOnramp } from 'app/utils/useCoinbaseOnramp'
import { toNiceError } from 'app/utils/toNiceError'
import { Button, FadeCard, Paragraph, Spinner, YStack } from '@my/ui'
import { DepositConfirm } from 'app/features/deposit/components/DepositConfirm'
import { IconError } from 'app/components/icons'
import { useThemeSetting } from '@tamagui/next-theme'

const COINBASE_APP_ID = process.env.NEXT_PUBLIC_CDP_APP_ID ?? ''

interface DepositCoinbaseScreenProps {
  defaultPaymentMethod?: 'APPLE_PAY' | 'CARD'
}

export function DepositCoinbaseScreen({ defaultPaymentMethod }: DepositCoinbaseScreenProps) {
  const { data: sendAccount } = useSendAccount()
  const { openOnramp, closeOnramp, status, error, isLoading } = useCoinbaseOnramp({
    projectId: COINBASE_APP_ID,
    address: sendAccount?.address ?? '',
    partnerUserId: sendAccount?.user_id ?? '',
    defaultPaymentMethod,
  })
  const { resolvedTheme } = useThemeSetting()
  const isDarkTheme = resolvedTheme?.startsWith('dark')

  const handleConfirmTransaction = (amount: number) => {
    openOnramp(amount)
  }

  const renderContent = () => {
    switch (true) {
      case !!error:
        return (
          <FadeCard ai={'center'}>
            <IconError size={'$4'} color={'$error'} />
            <YStack ai={'center'} gap={'$2'}>
              <Paragraph size={'$8'} fontWeight={500} ta={'center'}>
                Unable to Initialize Payment
              </Paragraph>
              <Paragraph
                size={'$5'}
                ta={'center'}
                color={'$lightGrayTextField'}
                $theme-light={{ color: '$darkGrayTextField' }}
              >
                {toNiceError(error)}
              </Paragraph>
            </YStack>
            <Button theme="green" py={'$5'} br={'$4'} mt={'$4'} onPress={closeOnramp} w={'100%'}>
              <Button.Text
                ff={'$mono'}
                fontWeight={'500'}
                tt="uppercase"
                size={'$5'}
                color={'$black'}
              >
                try again
              </Button.Text>
            </Button>
          </FadeCard>
        )
      case status === 'success':
        return (
          <FadeCard ai={'center'}>
            <Spinner size="large" color={isDarkTheme ? '$primary' : '$color12'} />
            <YStack ai={'center'} gap={'$2'}>
              <Paragraph size={'$8'} fontWeight={500} $gtLg={{ size: '$9' }} ta={'center'}>
                Almost there...
              </Paragraph>
              <Paragraph
                ta={'center'}
                size={'$5'}
                color={'$lightGrayTextField'}
                $theme-light={{ color: '$darkGrayTextField' }}
              >
                Just wrapping up the final steps.
              </Paragraph>
            </YStack>
          </FadeCard>
        )
      case status === 'pending':
        return (
          <FadeCard ai={'center'}>
            <Spinner size="large" color={isDarkTheme ? '$primary' : '$color12'} />
            <YStack ai={'center'} gap={'$2'}>
              <Paragraph size={'$8'} fontWeight={500} ta={'center'} $gtLg={{ size: '$9' }}>
                Hold tight...
              </Paragraph>
              <Paragraph
                ta={'center'}
                size={'$5'}
                color={'$lightGrayTextField'}
                $theme-light={{ color: '$darkGrayTextField' }}
              >
                Complete the transaction in your Coinbase window.
              </Paragraph>
            </YStack>
            <Button
              transparent
              chromeless
              backgroundColor="transparent"
              hoverStyle={{ backgroundColor: 'transparent' }}
              pressStyle={{ backgroundColor: 'transparent' }}
              focusStyle={{ backgroundColor: 'transparent' }}
              p={0}
              bw={0}
              height={'auto'}
              onPress={closeOnramp}
            >
              <Button.Text
                ff={'$mono'}
                fontWeight={'500'}
                tt="uppercase"
                size={'$5'}
                hoverStyle={{ color: '$primary' }}
              >
                cancel
              </Button.Text>
            </Button>
          </FadeCard>
        )
      case status === 'failed':
        return (
          <FadeCard ai={'center'}>
            <IconError size={'$4'} color={'$error'} />
            <YStack ai={'center'} gap={'$2'}>
              <Paragraph size={'$8'} fontWeight={500} ta={'center'} $gtLg={{ size: '$9' }}>
                Payment Failed
              </Paragraph>
              <Paragraph
                size={'$5'}
                ta={'center'}
                color={'$lightGrayTextField'}
                $theme-light={{ color: '$darkGrayTextField' }}
              >
                Please check your connection or try a different card.
              </Paragraph>
            </YStack>
            <Button theme="green" py={'$5'} br={'$4'} mt={'$4'} onPress={closeOnramp} w={'100%'}>
              <Button.Text
                ff={'$mono'}
                fontWeight={'500'}
                tt="uppercase"
                size={'$5'}
                color={'$black'}
              >
                try again
              </Button.Text>
            </Button>
          </FadeCard>
        )
      default:
        return (
          <DepositConfirm onConfirmTransaction={handleConfirmTransaction} isLoading={isLoading} />
        )
    }
  }

  return (
    <YStack
      w={'100%'}
      gap="$5"
      py={'$3.5'}
      $gtLg={{
        w: '50%',
      }}
    >
      {renderContent()}
    </YStack>
  )
}
