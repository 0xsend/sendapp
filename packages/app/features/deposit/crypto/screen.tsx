import { useSendAccount } from 'app/utils/send-accounts'
import { DepositAddressQR } from 'app/features/deposit/components/DepositAddressQR'
import { Button, FadeCard, Paragraph, Spinner, useToastController, YStack } from '@my/ui'
import { Check } from '@tamagui/lucide-icons'
import * as Clipboard from 'expo-clipboard'
import { useState } from 'react'
import { IconCopy } from 'app/components/icons'
import { CopyAddressDialog } from 'app/features/deposit/components/CopyAddressDialog'
import { Platform } from 'react-native'

export function DepositCryptoScreen() {
  const { data: sendAccount, isLoading: isLoadingSendAccount } = useSendAccount()
  const toast = useToastController()
  const [hasCopied, setHasCopied] = useState(false)
  const [isConfirmed, setIsConfirmed] = useState(false)
  const [copyAddressDialogIsOpen, setCopyAddressDialogIsOpen] = useState(false)

  const copyToClipboard = async () => {
    await Clipboard.setStringAsync(sendAccount?.address || '').catch(() =>
      toast.show('Something went wrong', {
        message: 'We were unable to copy your address to the clipboard',
        customData: {
          theme: 'red',
        },
      })
    )
    setHasCopied(true)
  }

  const copyOnPress = async () => {
    if (!isConfirmed) {
      setCopyAddressDialogIsOpen(true)
      return
    }

    void copyToClipboard()
  }

  if (isLoadingSendAccount) return <Spinner size="large" />

  return (
    <YStack
      w={'100%'}
      gap="$5"
      py={'$3.5'}
      ai={'center'}
      $gtLg={{
        w: '50%',
        ai: 'flex-start',
      }}
    >
      <YStack gap={'$3.5'}>
        <FadeCard>
          <Paragraph size={'$7'}>Base Network</Paragraph>
          {Platform.OS === 'web' && (
            <DepositAddressQR
              address={sendAccount?.address}
              onPress={() => setCopyAddressDialogIsOpen(true)}
              isConfirmed={isConfirmed}
            />
          )}
        </FadeCard>
      </YStack>
      <YStack width={'100%'}>
        <YStack
          marginHorizontal={'auto'}
          maxWidth={'100%'}
          gap={'$3.5'}
          $gtLg={{ marginHorizontal: 0 }}
        >
          <Paragraph w={'100%'} size={'$7'}>
            Wallet Address
          </Paragraph>
          <FadeCard maxWidth={'100%'}>
            <Button
              chromeless
              backgroundColor="transparent"
              hoverStyle={{ backgroundColor: 'transparent' }}
              pressStyle={{
                backgroundColor: 'transparent',
                borderColor: 'transparent',
              }}
              focusStyle={{ backgroundColor: 'transparent' }}
              p={0}
              height={'auto'}
              onPress={copyOnPress}
            >
              <Button.Text fontSize={'$5'}>{sendAccount?.address}</Button.Text>
              <Button.Icon>
                {hasCopied ? (
                  <Check color="$primary" size="$1" $theme-light={{ color: '$color12' }} />
                ) : (
                  <IconCopy
                    flexShrink={0}
                    size="$1"
                    $theme-dark={{ color: '$primary' }}
                    $theme-light={{ color: '$color12' }}
                  />
                )}
              </Button.Icon>
            </Button>
          </FadeCard>
        </YStack>
      </YStack>
      <CopyAddressDialog
        isOpen={copyAddressDialogIsOpen}
        onClose={() => {
          setCopyAddressDialogIsOpen(false)
        }}
        onConfirm={() => {
          setIsConfirmed(true)
          setCopyAddressDialogIsOpen(false)
          void copyToClipboard()
        }}
      />
    </YStack>
  )
}
