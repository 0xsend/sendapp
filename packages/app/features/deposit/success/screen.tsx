import { CheckCircle } from '@tamagui/lucide-icons'
import { FadeCard, Paragraph, PrimaryButton, YStack } from '@my/ui'
import { Platform } from 'react-native'
import { useLink } from 'solito/link'

export function DepositSuccessScreen() {
  const linkProps = useLink({ href: Platform.OS === 'web' ? '/deposit' : '/(tabs)/' })

  return (
    <YStack
      w={'100%'}
      gap="$5"
      py={'$3.5'}
      jc={'space-between'}
      $gtLg={{
        w: '50%',
      }}
    >
      <FadeCard ai={'center'}>
        <CheckCircle size={'$3'} color={'$primary'} $theme-light={{ color: '$color12' }} />
        <YStack ai={'center'} gap={'$2'}>
          <Paragraph size={'$8'} fontWeight={500} ta={'center'} $gtLg={{ size: '$9' }}>
            Deposit Complete
          </Paragraph>
          <Paragraph
            size={'$5'}
            ta={'center'}
            color={'$lightGrayTextField'}
            $theme-light={{ color: '$darkGrayTextField' }}
          >
            You’re all set — your funds will appear in your wallet soon
          </Paragraph>
        </YStack>
        <PrimaryButton {...linkProps}>
          <PrimaryButton.Text>
            {Platform.OS === 'web' ? 'deposit again' : 'continue'}
          </PrimaryButton.Text>
        </PrimaryButton>
      </FadeCard>
    </YStack>
  )
}
