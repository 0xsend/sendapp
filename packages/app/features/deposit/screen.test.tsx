import '@jest/globals'
import { render, screen, waitFor } from '@testing-library/react-native'
import { Provider } from 'app/__mocks__/app/provider'
import { DepositScreen } from './screen'

jest.mock('wagmi')

jest.mock('@my/ui', () => ({
  ...jest.requireActual('@my/ui'),
  Fade: ({ children }) => children,
}))

describe('DepositScreen', () => {
  it('renders the deposit screen', async () => {
    render(
      <Provider>
        <DepositScreen />
      </Provider>
    )

    // screen.debug({ message: 'DepositScreen: render' })
    await waitFor(() => expect(screen.getByText('Deposit on Base')).toBeVisible())
    expect(screen).toMatchSnapshot()
  })
})
