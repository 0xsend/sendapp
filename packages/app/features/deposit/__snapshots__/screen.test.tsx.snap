// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositScreen renders the deposit screen 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "gap": 24,
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 7,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 36,
          "fontWeight": 500,
          "lineHeight": 35,
          "userSelect": "auto",
        }
      }
      suppressHighlighting={true}
    >
      Deposit Funds
    </Text>
    <Text
      style={
        {
          "color": "#B3B3B3",
          "fontFamily": "System",
          "fontSize": 19.2,
          "fontWeight": "400",
          "lineHeight": 19,
          "userSelect": "auto",
        }
      }
      suppressHighlighting={true}
    >
      Select a deposit method
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 16,
        "width": "100%",
      }
    }
  >
    <View
      href="/deposit/crypto"
      testID="MockSolitoLink"
    >
      <View
        collapsable={false}
        forwardedRef={[Function]}
        jestAnimatedStyle={
          {
            "value": {
              "opacity": 0,
              "transform": [
                {
                  "scale": 0.9,
                },
              ],
            },
          }
        }
        style={
          {
            "opacity": 0,
            "transform": [
              {
                "scale": 0.9,
              },
            ],
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#111f22",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexDirection": "column",
              "gap": 16,
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "gap": 24,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={28}
                bbWidth={28}
                color="#40FB50"
                focusable={false}
                height={28}
                meetOrSlice={0}
                minX={0}
                minY={0}
                strokeWidth="2"
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "color": "#40FB50",
                      "flexShrink": 0,
                    },
                    {
                      "flex": 0,
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
                tintColor="#40FB50"
                vbHeight={24}
                vbWidth={24}
                width={28}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "strokeWidth",
                    ]
                  }
                  strokeWidth="2"
                >
                  <RNSVGPath
                    d="M2 20V4H22V20H2ZM4 8H20V6H4V8ZM15.775 14.075L20 10.525V10H4V11.225L15.775 14.075Z"
                    fill={
                      {
                        "type": 2,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "column",
                    "gap": 7,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 21.599999999999998,
                      "fontWeight": 500,
                      "lineHeight": 19,
                      "userSelect": "auto",
                    }
                  }
                  suppressHighlighting={true}
                >
                  Crypto Wallet
                </Text>
                <Text
                  style={
                    {
                      "color": "#B3B3B3",
                      "fontFamily": "System",
                      "fontSize": 19.2,
                      "fontWeight": "400",
                      "lineHeight": 19,
                      "userSelect": "auto",
                      "width": "100%",
                    }
                  }
                  suppressHighlighting={true}
                >
                  Deposit from an external wallet
                </Text>
              </View>
            </View>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="none"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              stroke="#B3B3B3"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#B3B3B3",
                  },
                  {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              tintColor="#B3B3B3"
              vbHeight={24}
              vbWidth={24}
              width={20}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4289967027,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              >
                <RNSVGPath
                  d="m9 18 6-6-6-6"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4289967027,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
    <View
      href="/deposit/apple-pay"
      testID="MockSolitoLink"
    >
      <View
        collapsable={false}
        forwardedRef={[Function]}
        jestAnimatedStyle={
          {
            "value": {
              "opacity": 0,
              "transform": [
                {
                  "scale": 0.9,
                },
              ],
            },
          }
        }
        style={
          {
            "opacity": 0,
            "transform": [
              {
                "scale": 0.9,
              },
            ],
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#111f22",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexDirection": "column",
              "gap": 16,
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "gap": 24,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={28}
                bbWidth={28}
                color="#40FB50"
                focusable={false}
                height={28}
                meetOrSlice={0}
                minX={0}
                minY={0}
                strokeWidth="2"
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "color": "#40FB50",
                      "flexShrink": 0,
                    },
                    {
                      "flex": 0,
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
                tintColor="#40FB50"
                vbHeight={24}
                vbWidth={24}
                width={28}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "strokeWidth",
                    ]
                  }
                  strokeWidth="2"
                >
                  <RNSVGPath
                    d="M18.1726 22.1173C16.975 23.2783 15.6674 23.095 14.4086 22.5451C13.0766 21.9829 11.8545 21.9585 10.4491 22.5451C8.68937 23.3027 7.7606 23.0828 6.70962 22.1173C0.745927 15.9703 1.62582 6.60929 8.39607 6.26711C10.0459 6.35266 11.1946 7.17144 12.16 7.24477C13.6021 6.95147 14.983 6.10824 16.5228 6.21823C18.3681 6.36488 19.7613 7.09812 20.6779 8.41795C16.865 10.7032 17.7693 15.7259 21.2644 17.1313C20.5679 18.9644 19.6635 20.7853 18.1604 22.1296L18.1726 22.1173ZM12.0378 6.19379C11.8545 3.46858 14.0665 1.21997 16.6084 1C16.9628 4.15294 13.7487 6.49931 12.0378 6.19379Z"
                    fill={
                      {
                        "type": 2,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "column",
                    "gap": 7,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 21.599999999999998,
                      "fontWeight": 500,
                      "lineHeight": 19,
                      "userSelect": "auto",
                    }
                  }
                  suppressHighlighting={true}
                >
                  Apple Pay
                </Text>
                <Text
                  style={
                    {
                      "color": "#B3B3B3",
                      "fontFamily": "System",
                      "fontSize": 19.2,
                      "fontWeight": "400",
                      "lineHeight": 19,
                      "userSelect": "auto",
                      "width": "100%",
                    }
                  }
                  suppressHighlighting={true}
                >
                  Debit card only
                </Text>
              </View>
            </View>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="none"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              stroke="#B3B3B3"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#B3B3B3",
                  },
                  {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              tintColor="#B3B3B3"
              vbHeight={24}
              vbWidth={24}
              width={20}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4289967027,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              >
                <RNSVGPath
                  d="m9 18 6-6-6-6"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4289967027,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
    <View
      href="/deposit/debit-card"
      testID="MockSolitoLink"
    >
      <View
        collapsable={false}
        forwardedRef={[Function]}
        jestAnimatedStyle={
          {
            "value": {
              "opacity": 0,
              "transform": [
                {
                  "scale": 0.9,
                },
              ],
            },
          }
        }
        style={
          {
            "opacity": 0,
            "transform": [
              {
                "scale": 0.9,
              },
            ],
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#111f22",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexDirection": "column",
              "gap": 16,
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "gap": 24,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={28}
                bbWidth={28}
                color="#40FB50"
                focusable={false}
                height={28}
                meetOrSlice={0}
                minX={0}
                minY={0}
                strokeWidth="2"
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "color": "#40FB50",
                      "flexShrink": 0,
                    },
                    {
                      "flex": 0,
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
                tintColor="#40FB50"
                vbHeight={24}
                vbWidth={24}
                width={28}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "strokeWidth",
                    ]
                  }
                  strokeWidth="2"
                >
                  <RNSVGPath
                    d="M2 20V4H22V20H2ZM4 8H20V6H4V8ZM4 18H20V12H4V18Z"
                    fill={
                      {
                        "type": 2,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "column",
                    "gap": 7,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 21.599999999999998,
                      "fontWeight": 500,
                      "lineHeight": 19,
                      "userSelect": "auto",
                    }
                  }
                  suppressHighlighting={true}
                >
                  Debit Card
                </Text>
                <Text
                  style={
                    {
                      "color": "#B3B3B3",
                      "fontFamily": "System",
                      "fontSize": 19.2,
                      "fontWeight": "400",
                      "lineHeight": 19,
                      "userSelect": "auto",
                      "width": "100%",
                    }
                  }
                  suppressHighlighting={true}
                >
                  Quick deposits up to $500/week
                </Text>
              </View>
            </View>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="none"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              stroke="#B3B3B3"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#B3B3B3",
                  },
                  {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              tintColor="#B3B3B3"
              vbHeight={24}
              vbWidth={24}
              width={20}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4289967027,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              >
                <RNSVGPath
                  d="m9 18 6-6-6-6"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4289967027,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
